WASI FileExt::tell is gone, breaks build

Hi,

building `cap-std` against `wasm32-wasip1` fails with the following compiler error with recent nightlies:

```
error[E0782]: expected a type, found a trait
   --> cap-std/src/fs/file.rs:528:9
    |
528 |         std::os::wasi::fs::FileExt::tell(&self.std)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: you can add the `dyn` keyword if you want a trait object
    |
528 |         <dyn std::os::wasi::fs::FileExt>::tell(&self.std)
    |         ++++                           +

For more information about this error, try `rustc --explain E0782`.
```

The method was removed here: https://github.com/rust-lang/rust/pull/137205

(I'm not sure how cap-std handles version bumps considering that WASI support depends on the unstable `wasi_ext` feature but I guess just removing the `fn tell(&self)` from `cap-std`'s extension trait might be fine?)

Thanks! 🙂

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -371,6 +371,10 @@ jobs:
     - uses: ./.github/actions/install-rust
       with:
         toolchain: ${{ matrix.rust }}
+
+    - name: Use specific dependency versions for Rust 1.63 compatibility.
+      run: cargo update --package=once_cell --precise=1.20.3
+
     # Don't use --all-features because some of the features have dependencies
     # that don't work on newer Rust versions.
     - run: cargo test --workspace --features=fs_utf8,arf_strings

--- a/cap-primitives/src/fs/file.rs
+++ b/cap-primitives/src/fs/file.rs
@@ -157,9 +157,6 @@ pub trait FileExt {
         Ok(())
     }
 
-    /// Returns the current position within the file.
-    fn tell(&self) -> io::Result<u64>;
-
     /// Adjust the flags associated with this file.
     fn fdstat_set_flags(&self, flags: u16) -> io::Result<()>;
 

--- a/cap-std/src/fs/file.rs
+++ b/cap-std/src/fs/file.rs
@@ -524,11 +524,6 @@ impl crate::fs::FileExt for File {
     }
 
     #[inline]
-    fn tell(&self) -> std::result::Result<u64, io::Error> {
-        std::os::wasi::fs::FileExt::tell(&self.std)
-    }
-
-    #[inline]
     fn fdstat_set_flags(&self, flags: u16) -> std::result::Result<(), io::Error> {
         std::os::wasi::fs::FileExt::fdstat_set_flags(&self.std, flags)
     }

--- a/cap-std/src/fs_utf8/file.rs
+++ b/cap-std/src/fs_utf8/file.rs
@@ -522,11 +522,6 @@ impl crate::fs::FileExt for File {
     }
 
     #[inline]
-    fn tell(&self) -> std::result::Result<u64, io::Error> {
-        self.cap_std.tell()
-    }
-
-    #[inline]
     fn fdstat_set_flags(&self, flags: u16) -> std::result::Result<(), io::Error> {
         self.cap_std.fdstat_set_flags(flags)
     }

